app/models/service_orchestration.rb

```diff
@@ -99,19 +100,33 @@ def assign_vms_owner
   end
 
   def build_stack_create_options
-    # manager from dialog_options overrides the one copied from service_template
     dialog_options = options[:dialog] || {}
-    manager_from_dialog = OptionConverter.get_manager(dialog_options)
-    self.orchestration_manager = manager_from_dialog if manager_from_dialog
-    raise _("orchestration manager was not set") if orchestration_manager.nil?
-
-    # orchestration template from dialog_options overrides the one copied from service_template
-    template_from_dialog = OptionConverter.get_template(dialog_options)
-    self.orchestration_template = template_from_dialog if template_from_dialog
+    pick_orchestration_manager(dialog_options)
+    pick_orchestration_template(dialog_options)
 
     build_stack_options_from_dialog(dialog_options)
   end
 
+  # The order to pick the orchestration manager and template
+  # 1. The ones directly set through setter
+  # 2. The ones set through dialog options
+  # 3. The ones copied from service_template
+  def pick_orchestration_manager(dialog_options)
+    if orchestration_manager == service_template.orchestration_manager
```

Comment:
Ah, ok I am reading it badly. orchestration_manager == service_template.orchestration_manager
 means you didn't change it in custom automate script